backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -365,4 +418,362 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
+          .setDateUpdated(Calendar.getInstance().getTimeInMillis());
+    }
+
+    if (request.getCodeVersionSnapshot() != null) {
+      experimentRunBuilder.setCodeVersionSnapshot(request.getCodeVersionSnapshot());
+    }
+    if (request.getVersionedInputs() != null && request.hasVersionedInputs()) {
+      experimentRunBuilder.setVersionedInputs(request.getVersionedInputs());
+    }
+    if (userInfo != null) {
+
+      experimentRunBuilder.setOwner(userInfo.getVertaInfo().getUserId());
+    }
+
+    return InternalFuture.completedInternalFuture(experimentRunBuilder.build());
+  }
+
+  private InternalFuture<ExperimentRun> insertExperimentRun(ExperimentRun newExperimentRun) {
+    final var now = Calendar.getInstance().getTimeInMillis();
+    return jdbi.withHandle(
+            handle -> {
+              Map<String, Object> runValueMap = new LinkedHashMap<>();
+              runValueMap.put("id", newExperimentRun.getId());
+              runValueMap.put("project_id", newExperimentRun.getProjectId());
+              runValueMap.put("experiment_id", newExperimentRun.getExperimentId());
+              runValueMap.put("name", newExperimentRun.getName());
+              runValueMap.put("description", newExperimentRun.getDescription());
+              runValueMap.put("date_created", newExperimentRun.getDateCreated());
+              runValueMap.put("date_updated", newExperimentRun.getDateUpdated());
+              runValueMap.put("start_time", newExperimentRun.getStartTime());
+              runValueMap.put("end_time", newExperimentRun.getEndTime());
+              runValueMap.put("code_version", newExperimentRun.getCodeVersion());
+              // TODO: code version snapshot
+              /*runValueMap.put("code_version_snapshot_id", newExperimentRun.getCodeVersionSnapshot());*/
+              runValueMap.put("job_id", newExperimentRun.getJobId());
+              runValueMap.put("parent_id", newExperimentRun.getParentId());
+              runValueMap.put("owner", newExperimentRun.getOwner());
+
+              EnvironmentBlob environmentBlob =
+                  sortPythonEnvironmentBlob(newExperimentRun.getEnvironment());
+              runValueMap.put(
+                  "environment", ModelDBUtils.getStringFromProtoObject(environmentBlob));
+              runValueMap.put("deleted", false);
+
+              String[] fieldsArr = runValueMap.keySet().toArray(new String[0]);
+              String commaFields = String.join(",", fieldsArr);
+
+              StringBuilder queryStrBuilder =
+                  new StringBuilder("insert into experiment_run ( ")
+                      .append(commaFields)
+                      .append(") values (");
+
+              for (int i = 0; i < fieldsArr.length; i++) {
+                queryStrBuilder.append(":").append(fieldsArr[i]);
+                if (i < fieldsArr.length - 1) {
+                  queryStrBuilder.append(",");
+                }
+              }
+              queryStrBuilder.append(" ) ");
```

Comment:
This SQL command is very hard to read, which means it's unmaintainable and has a bug for all I care. We need to make this more readable by humans.

Reply:
Improve for loop with inline labda and simplify it